vendor/golang.org/x/sys/unix/zsyscall_openbsd_ppc64.go

```diff
@@ -527,6 +527,14 @@ func ioctl(fd int, req uint, arg uintptr) (err error) {
 	return
 }
 
+func ioctlPtr(fd int, req uint, arg unsafe.Pointer) (err error) {
+	_, _, e1 := syscall_syscall(libc_ioctl_trampoline_addr, uintptr(fd), uintptr(req), uintptr(arg))
+	if e1 != 0 {
+		err = errnoErr(e1)
+	}
+	return
+}
+
 var libc_ioctl_trampoline_addr uintptr
 
 //go:cgo_import_dynamic libc_ioctl ioctl "libc.so"
```

Comment:
Based on the provided code patch, here are my observations:

1. A new function `ioctlPtr` has been added which takes a void pointer `arg` instead of an integer argument.

2. The `ioctlPtr` function is implemented using the `lib_syscall_trampoline_addr` symbol and the `syscall.Syscall` function to call the `ioctl` system call.

3. The `err` variable is assigned the value returned by the `errnoErr` function if `e1` (return value of `syscall.Syscall`) is non-zero.

4. No comments or documentation have been added to describe the purpose of the new function or its parameters.

5. It is not clear from the code what problem the new function is trying to solve or what benefit it provides over the existing `ioctl` function.

6. It would be useful to perform testing of the new function to ensure that it does not introduce any regressions or compatibility issues with other parts of the codebase.

7. Possible improvement suggestion may include adding comments or documentation explaining the new function's purpose, documenting expected behavior and return values, and considering adding test cases for the function.